build: install the modulemap

compnerd opened the pull request:
The Swift fork has introduced a modulemap which is expected for some usage (primarily swift-markdown).  This adds the modulemap to the installation set.

compnerd commented:
CC: @QuietMisdreavus @etcwilde

etcwilde reviewed (approved):
Seems reasonable for things consuming cmark from Swift.

compnerd committed:
build: install additional headers

Add additional headers that are required for the GFM extensions library.

compnerd committed:
build: split apart installed headers and install module.modulemap

Add the modulemap for cmark_gfm_extensions and split up the installed
headers as there are two separate modulemaps.  This should hopefully
enable us to build swift-markdown.
